slickml/optimization.py

```diff
@@ -335,3 +339,158 @@ def plot_optimization_results(self):
+
+        if isinstance(metrics, str):
+            self.metrics = metrics
+        else:
+            raise TypeError("The input metrics must be a str")
+
+        if isinstance(num_boost_rounds, int):
+            self.num_boost_rounds = num_boost_rounds
+        else:
+            raise TypeError("The input num_boost_rouns must be a int")
+
+        if isinstance(n_split, int):
+            self.n_split = n_split
+        else:
+            raise TypeError("The input n_split must be a int")
+
+        if isinstance(shuffle, bool):
+            self.shuffle = shuffle
+        else:
+            raise TypeError("The input shuffle must be a boolean")
+
+        if isinstance(early_stop, int):
+            self.early_stop = early_stop
+        else:
+            raise TypeError("The input early_stop must be a int")
+
+        if isinstance(verbose, bool):
+            self.verbose = verbose
+        else:
+            raise TypeError("The input verbose must be a boolean")
+
+    def process(self, X_train, Y_train, func_name, space, trials, algo, max_evals):
+        """
+        Explore a function over a hyperparameter space
+        according to a given algorithm, allowing up to a certain number of
+        function evaluations.
+        Parameters
+        ----------
+        X_train: numpy.array or Pandas DataFrame
+            Training features data
+        y_train: numpy.array[int] or list[int]
+            List of training ground truth binary values [0, 1]
+        func_name: str
+            function name for performing optimization
+        space: dict
+            The set of possible arguments to `fn` is the set of objects
+            that could be created with non-zero probability by drawing randomly
+            from this stochastic program involving involving hp
+        trials: object
+            Storage for completed, ongoing, and scheduled evaluation points.  If
+            None, then a temporary `base.Trials` instance will be created.  If
+            a trials object, then that trials object will be affected by
+            side-effect of this call
+        algo: object
+            provides logic for sequential search of the hyperparameter space
+        max_evals: int
+            Storage for completed, ongoing, and scheduled evaluation points.  If
+            None, then a temporary `base.Trials` instance will be created.  If
+            a trials object, then that trials object will be affected by
+            side-effect of this call
+        """
+
+        fn = getattr(self, func_name)
+        self.X_train = X_train
+        self.y_train = Y_train
+        try:
+            results = fmin(
+                fn, space=space, algo=algo, max_evals=max_evals, trials=trials
+            )
+            results = pd.DataFrame(results, index=[0])
+        except Exception as e:
+            return {"status": STATUS_FAIL, "exception": str(e)}
+        return results, trials
```

Comment:
fit function does not return anything. Therefore helper functions needed to return the outputs.
like get_optimization_results() and get_optimization_trials(). Please do check in Bayesian. So, it will follow the same naming convention.